upstream.sh

```diff
@@ -11,6 +11,10 @@ PROJECT_NAME=$(basename `git rev-parse --show-toplevel`)
 echo $PROJECT_NAME
 COVERAGE_SHA=$(git rev-list HEAD | head -1)
 echo $COVERAGE_SHA > SHA.txt
+echo "Processing coverage for report belonging to latest commit"
+PROD_COVERAGE=$(curl https://codecov.io/api/gh/codecov/$PROJECT_NAME/commit/$COVERAGE_SHA | \
+python2 -c 'import json,sys;obj=json.load(sys.stdin);print obj["commit"]["totals"]["c"]')
```

Comment:
I mused over this one for a long time. My first thought was it was weird to bring in python2 to extract one key from a json object, but I understand the shell makes json manipulation pretty tough. 

Since you only need  one value though, and will likely not have to do a lot of json manipulation outside of that, and you're working with a really simple value for the key (essentially a number), a built in shell tool like grep _may_ be enough. Something like:

```
grep -Po '"text":.*?[^\\]",' tweets.json
```
(Source: https://stackoverflow.com/questions/1955505/parsing-json-with-unix-tools)

I don't think I'd block a PR on this, because the python choice is likely good enough (although I'd probably do this with python 3 since 2 is fully deprecated and no longer supported at this point) and a recommended answer from even the source I linked, but I think it's worth experimenting with grep, etc a bit to see if you can extract this value using only shell native tools. I recommend this only because what you need to do with the json is sufficiently simple, if your use case was more complex, I think I'd recommend `jq` or `python` etc.

Reply:
After reading the article I opted for using Python (now upgraded to python3). My reasoning was that Python is already installed in the GHA Ubuntu machine so there isn't an extra dependency that we'd be installing. I meddled with grep for a bit but read object manipulation in general is better suited for a language that can handle JSON easily, with little expense since Python is already installed. Happy to discuss the alternative if we think it's a better route.

Reply:
This suits me. Thanks for taking a look